trikot-viewmodels-declarative-flow/viewmodels-declarative-flow/src/commonMain/kotlin/com/mirego/trikot/viewmodels/declarative/viewmodel/VMDLifecycleViewModelImpl.kt

```diff
@@ -0,0 +1,45 @@
+package com.mirego.trikot.viewmodels.declarative.viewmodel
+
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+
+abstract class VMDLifecycleViewModelImpl(coroutineScope: CoroutineScope) : VMDLifecycleViewModel, VMDViewModelImpl(coroutineScope) {
+
+    private var firstOnAppear = true
+    var viewLifecycleCoroutineScope: CoroutineScope? = null
+        private set
+
+    final override fun onAppear() {
+        viewLifecycleCoroutineScope?.cancel()
+        viewLifecycleCoroutineScope = CoroutineScope(
+            Dispatchers.Main.immediate + SupervisorJob() + CoroutineExceptionHandler { _, exception ->
```

Comment:
Why don't we use CoroutineScopeProvider here?

Reply:
Initially, it was because we add an Exception handler. I will add a function in the provider that takes an exceptionHandler as argument.

The scope provider was initially built to be able to provide a scope in Swift code, but I see it can provide also elsewhere.